connection.go

```diff
@@ -581,43 +591,63 @@ func pack(h *smallWBuf, enc *encoder, reqid uint32,
 	return
 }
 
-func (conn *Connection) writeAuthRequest(w *bufio.Writer, scramble []byte) (err error) {
+func (conn *Connection) writeRequest(w *bufio.Writer, req Request,
+	reqName string) (err error) {
 	var packet smallWBuf
-	req := newAuthRequest(conn.opts.User, string(scramble))
 	err = pack(&packet, newEncoder(&packet), 0, req, ignoreStreamId, conn.Schema)
 
 	if err != nil {
-		return errors.New("auth: pack error " + err.Error())
+		return errors.New(reqName + ": pack error " + err.Error())
```

Comment:
Here you can not use the argument reqName, but wrap the error in a caller code https://go.dev/blog/go1.13-errors , see `Wrapping errors with %w`

The result will be something like that:

```Go
if err := writeRequest(w, req); err != nil {
    return fmt.Errorf("failed to send iproto id: %w", err)
}
```

Reply:
Thank you, it's much better. Fixed.